subprojects/dependency-management/src/main/java/org/gradle/internal/component/local/model/MissingLocalArtifactMetadata.java

```diff
@@ -98,4 +98,9 @@ public boolean equals(Object obj) {
         MissingLocalArtifactMetadata other = (MissingLocalArtifactMetadata) obj;
         return other.componentIdentifier.equals(componentIdentifier) && other.name.equals(name);
     }
+
+    @Override
+    public boolean isOptionalArtifact() {
+        return true;
```

Comment:
This is not a method I am super familiar with.
Can you explain why a blanket change like this to all instances of `MissingLocalArtifactMetadata` is not an issue?

Reply:
So my previous test run [succeeded here](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/59703718?buildTab=overview&hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true), then my only change after was to add a new integration test.  So that's a little strange.

I was looking for failing tests and when I didn't see any assumed we could mark missing but local metadata artifacts as optional so that we don't throw an exception when trying to `get()` them.  I'll look a little deeper into these now failing tests, maybe they can throw some new light on this.